anillo/middlewares/json.py

```diff
@@ -33,17 +33,23 @@ def wrapper(request, *args, **kwargs):
     return wrapper
 
 
-def wrap_json_body(func):
+def wrap_json_body(func=None, *, preserve_raw_body=False):
     """
     A middleware that parses the body of json requests and
     add it to the request under the `body` attribute (replacing
-    the previous value).
+    the previous value). Can preserve the original value in
+    a new attribute `raw_body` if you give preserve_raw_body=True.
     """
 
+    if func is None:
+        return functools.partial(wrap_json_body, preserve_raw_body=preserve_raw_body)
+
     @functools.wraps(func)
     def wrapper(request, *args, **kwargs):
         ctype, pdict = parse_header(request.headers.get('Content-Type', ''))
         if ctype == "application/json":
+            if preserve_raw_body:
```

Comment:
Doing it in that way can make the use of raw_body property inconsistent. When request comes with json content type the raw body will be asigned, but if not, not. So in the application you should check always  for the content type before using that attribute, that seems like inconsistent.

Consider put that condition out of specific content type.